spec.md

```diff
@@ -1601,6 +1655,72 @@ The CO MUST implement the specified error recovery behavior when it encounters t
+
+`ControllerExpandVolume` RPC call MUST be idempotent and if size of underlying volume already meets requested capacity, the plugin MUST respond with successfull response.
+
+Examples:
+* Offline Volume Expansion:
+  Given a ElasticSearch process that runs on Azure Disk volume and needs more space.
+  - The administrator takes the elasticsearch server offline and request more space for volume from the CO.
+  - The CO in turn first makes `ControllerExpandVolume` RPC call which results in requesting more space from Azure cloudprovider for volume-id that was being used by ElasticSearch.
+  - Once `ControllerExpandVolume` finishes the CO will inform administrator about it and administrator will resume the ElasticSearch workload.
+  - On node where ElasticSearch workload is scheduled when workload starts after `NodeStageVolume` the CO calls `NodeExpandVolume`.
+  - Calling `NodeExpandVolume` on volume results in expanding the underlying file system and added space becomes available to workload when it startsup.
+* Online Volume Expansion:
+  Given a Mysql server running using Openstack Cinder and needs more space.
+  - The administrator requests more space for volume from the CO.
+  - The CO in turn first makes `ControllerExpandVolume` RPC call which results in requesting more space from Openstack Cinder for given volume.
+  - The node where mysql workload is running the CO makes `NodeExpandVolume` RPC call while volume is in-use using path where volume is staged.
+  - Calling `NodeExpandVolume` on volume results in expanding the underlying file system and added space automatically becomes available to mysql workload without any downtime.
+
+
+```protobuf
+message ControllerExpandVolumeRequest {
+  // The ID of the volume to expand. This field is REQUIRED.
+  string volume_id = 1;
+
+  // This field is REQUIRED. This allows CO to specify the
+  // capacity requirements of the volume after expansion.
+  CapacityRange capacity_range = 2;
+
+  // Secrets required by the plugin for expanding the volume.
+  // This field is OPTIONAL.
+  map<string, string> controller_expand_volume_secrets = 3;
+}
+
+message ControllerExpandVolumeResponse {
+  // Capacity of volume after expansion.
+  // This field is REQUIRED.
+  int64 capacity_bytes = 1;
+
+  // Whether file system expansion is required for the volume.
+  // When true the CO MUST make NodeExpandVolume RPC call
+  // on the node. This field is REQUIRED.
+  bool fs_resize_required = 2;
+}
+```
+
+##### ControllerExpandVolume Errors
+
+| Condition | gRPC Code | Description | Recovery Behavior |
+|-----------|-----------|-------------|-------------------|
+| Volume in use | 9 FAILED_PRECONDITION | Indicates that the volume corresponding to the specified `volume_id` could not be expanded because it is in use. | Caller SHOULD ensure that volume is not in-use and retry with exponential back off. |
```

Comment:
What do you mean by `in use` here? I'd be more specific: the volume is currently published on a node but the plugin does not support ONLINE expansion capability.